src/NativeShell/www/apphost.js

```diff
@@ -0,0 +1,172 @@
+//disables all console logs
+console.log = function () {};
+
+var deviceId;
+var deviceName;
+var appName;
+var appVersion;
+
+var features = [
+    //'filedownload',
+    'sync',
+    'customsyncpath',
+    'displaylanguage',
+    'subtitleappearancesettings',
+    'cameraupload',
+    'sharing',
+    'exit',
+    'htmlaudioautoplay',
+    'htmlvideoautoplay',
+    'externallinks',
+    'multiserver',
+    'physicalvolumecontrol',
+    'remotecontrol',
+    'castmenuhashchange'
+];
+
+function getDeviceProfile(profileBuilder) {
+    var profile = profileBuilder();
+
+    profile.DirectPlayProfiles.push({
+        // TODO investigate ac3 support
+        Container: "m4v,3gp,ts,mpegts,mov,xvid,vob,mkv,wmv,asf,ogm,ogv,m2v,avi,mpg,mpeg,mp4,webm,wtv",
+        Type: 'Video',
+        AudioCodec: 'aac,aac_latm,mp2,mp3,wma,dca,dts,pcm,PCM_S16LE,PCM_S24LE,opus,flac'
+    });
+
+    profile.CodecProfiles = profile.CodecProfiles.filter(function (i) {
+        return i.Type == 'Audio';
+    });
+
+    profile.SubtitleProfiles.push(
+        {
+            Format: 'ssa',
+            Method: 'External'
+        },
+        {
+            Format: 'ass',
+            Method: 'External'
+        }
+    );
+
+    profile.CodecProfiles.push({
+        Type: 'Video',
+        Container: 'avi',
+        Conditions: [
+            {
+                Condition: 'NotEqual',
+                Property: 'CodecTag',
+                Value: 'xvid'
+            }
+        ]
+    });
+
+    profile.CodecProfiles.push({
+        Type: 'Video',
+        Codec: 'h264',
+        Conditions: [
+        {
+            Condition: 'EqualsAny',
+            Property: 'VideoProfile',
+            Value: 'high|main|baseline|constrained baseline'
+        },
+        {
+            Condition: 'LessThanEqual',
+            Property: 'VideoLevel',
+            Value: '41'
+        }]
+    });
+
+    //profile.TranscodingProfiles.filter(function (p) {
+
+    //    return p.Type == 'Video' && p.Container == 'mkv';
+
+    //}).forEach(function (p) {
+
+    //    p.Container = 'ts';
+    //});
+
+    profile.TranscodingProfiles.filter(function (p) {
+
+        return p.Type == 'Video' && p.CopyTimestamps == true;
+
+    }).forEach(function (p) {
+        // Vlc doesn't seem to handle this well
+        p.CopyTimestamps = false;
+    });
+
+    profile.TranscodingProfiles.filter(function (p) {
+
+        return p.Type == 'Video' && p.VideoCodec == 'h264';
+
+    }).forEach(function (p) {
+
+        p.AudioCodec += ',ac3';
```

Comment:
hmm, this seems to contradict https://github.com/jellyfin/jellyfin-android/pull/109/files#diff-a8a438d40fbe89a5e778e751a58f9339R31

Reply:
yup, you are right, but don't know what's the reason for that. When i search for ac3 support in html5 video, it doesn't support. So it should be subject to a transcode to play well on the clients using html5 video tag.

Reply:
No, actually one is for DirectPlay profiling and this one is for Transcode profiling. Since video html5 does not support ac3, needs transcoding. I think this is the case here.

Reply:
Than that line I was referring to ("TODO: investigate ac3 support") has to be removed, right?

Reply:
yup, makes no sense keeping that todo, will remove it right now.